filesdk/index.js

```diff
@@ -0,0 +1,151 @@
+"use strict";
+
+const fs = require('fs');
+const q = require('q');
+const rimraf = require('rimraf');
+const extract = require('extract-zip')
+const targz = require('targz');
+const ncp = require('ncp').ncp;
+ncp.limit = 16;
+
+/* working */
+let readFile = (path) => {
+    let defer = q.defer();
+    fs.readFile(path, 'utf8', function (err, contents) {
+        if (err) return defer.reject(err);
+        return defer.resolve(contents);
+    });
+    return defer.promise;
+}
+/* working, overwrites or creates new file */
+let writeFile = (path, contents) => {
```

Comment:
A suggestion: Most of the file operations having same logic. It might be a good idea to write a function having the common logic. And call it from the SDK functions.